### PR TITLE
Minor optimization

### PR DESCRIPTION
**Pull request by p1r473**
**Summary of Changes**

- Splitting and Reading: The optimized version directly splits the input line into key and value using the IFS="=" and read command, eliminating the need for an intermediate array (tokens).
- Trimming Spaces: The optimized version uses parameter expansion to remove leading and trailing spaces from key, which is more efficient and avoids the need for external commands like sed.
- Quote Handling: Both versions handle quotes in the value using the (Q) parameter expansion.

**Benefits of Optimization**

- Efficiency: Directly reading and splitting the line into key and value is more efficient than first reading the entire line into a variable and then splitting it.
- Simplicity: The optimized version is simpler and more readable, as it avoids the intermediate array and unnecessary operations.
- Performance: By avoiding external commands and intermediate steps, the optimized version improves the script's performance.

`alias -g | sort:`
This remains the same, listing all global aliases and sorting them.

`while IFS="=" read -r key value; do:`
This reads each line of the sorted aliases into the variables key and value directly, using the equal sign (=) as the delimiter. IFS (Internal Field Separator) is set to = to split the line into key and value.

`key="${key## }" and key="${key%% }":`
These lines remove leading and trailing spaces from key respectively. ${key## } removes leading spaces, and ${key%% } removes trailing spaces.

`value="${(Q)value}":`
This remains the same as in the original lines, removing quotes from the value using the (Q) parameter expansion modifier.

### CODE REVIEW

File: you-should-use.plugin.zsh [hunk 1]
```diff
@@ -135,6 +135,13 @@ function _check_git_aliases() {
         git config --get-regexp "^alias\..+$" | sort | while read key value; do
             key="${key#alias.}"
 
+            # if for some reason, read does not split correctly, we
+            # detect that and manually split the key and value
+            if [[ -z "$value" ]]; then
+                value="${key#* }"
+                key="${key%% *}"
+            fi
```
Comment (MichaelAquilina) [inline you-should-use.plugin.zsh:143]: I think you might need to update your base branch? This looks like code that was recently merged into main

Comment (p1r473) [reply you-should-use.plugin.zsh:143]: Sure I'll try to figure out how

Comment (p1r473) [reply you-should-use.plugin.zsh]: Fixed

**Comment by MichaelAquilina** [comment]
Tested it out and works well. Thanks :)

**Pull request merged**